site/zenodo_rdm/metrics/api.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""ZenodoRDM Metrics API."""
+
+import calendar
+from datetime import datetime, timedelta
+
+import requests
+from flask import current_app
+from invenio_accounts.models import User
+from invenio_communities.communities.records.models import CommunityMetadata
+from invenio_files_rest.models import FileInstance
+from invenio_search import current_search_client
+from invenio_search.utils import build_alias_name
+from opensearchpy import Search
+
+from zenodo_rdm.metrics.proxies import current_metrics
+
+
+class ZenodoMetric(object):
+    """API class for Zenodo Metrics."""
+
+    @staticmethod
+    def get_data_transfer():
+        """Get file transfer volume in TB."""
+        time_range = {"gte": current_metrics.metrics_start_date.isoformat()}
+
+        search = (
+            Search(
+                using=current_search_client,
+                index=build_alias_name("stats-file-download"),
+            )
+            .filter(
+                "range",
+                timestamp=time_range,
+            )
+            .filter(
+                "term",
+                is_parent=False,
+            )
```

Comment:
Hey @slint, need some context for this query. I don't think we have this field in ES anymore, is there a different field I should consider? Maybe alternate query is `parent_recid` should exist?

Reply:
I might be that actually we're not calculating stats the same way for all version as we did in legacy. Which means that you might not need this filter anymore. 

The difference between legacy/RDM stats aggregations for a record with 3 versions, is that we were also indexing a document containing aggregated numbers for all versions of a record (which is not the case anymore).

I would test to be sure that if e.g. you have 3 records with 2 file downloads each, we don't "double-count" everything.

Reply:
> @slint I checked, if we do download all, then only we count all files otherwise each file download is individually added.

Ok, that sounds good. The download-all is a new feature in RDM, so it will only apply there once we start getting usage.